### PR TITLE
Endpoint security state v1

### PR DESCRIPTION
**Pull request by darinfisher**
First pull request for a new template. 
This will let the users watch the security status for each of their endpoints.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Start by reading [how to use a template](docs/use_a_template.md), then check eac
 | [Currency Exchange Rates](currency_exchange_rates/) | Visualize and analyze currency exchange rates using Quandl. | Wojciech Kocjan |
 | [Covid-19 in South America](InfluxDBv2_Covid19_SouthAmerica/) | Current data and graphs covering Covid-19 cases and deaths in South America | [Ignacio Van Droogenbroeck](https://github.com/xe-nvdk) |
 | [Docker](docker/) | Monitor your running docker containers. | [@russorat](https://github.com/russorat) |
+| [Endpoint Security State](endpoing-security-state/) | Watch endpoing authentication availability and certificate status | [Darin Fisher](https://github.com/darinfisher) |
```
Comment (russorat) [inline README.md:22]:
```suggestion
| [Endpoint Security State](endpoing-security-state/) | Watch endpoint authentication availability and certificate status | [Darin Fisher](https://github.com/darinfisher) |
```

File: endpoint-security-state/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Endpoint Security State Template
+
+This InfluxDB template works by connecting to secure endpoints and attempting to log in. Using the `http_response` and `x509_cert` Telegraf plugins, availability, authentication, and certificate information is collected.
+The dashboard displays the general availability for each endpoint along with the x509 certificate status and authentication state.
+
+![Endpoint Security State Dashboard Screenshot](endpoint-security-state.png)
+
+- "✅" Service is functioning as expected.
+- "🟡" Service needs attention. The certificate will expire in less than 30 day.
```
Comment (russorat) [inline endpoint-security-state/README.md:9]:
```suggestion
- "🟡" Service needs attention. The certificate will expire in less than 30 days.
```

File: endpoint-security-state/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    - Authentication is enabled
+    - Authentication is working
+- 1 Dashboard: `Endpoint Security State`
+- 1 Variable: `bucket`
+
+## Setup Instructions
+
+### Install the template package
+From your [InfluxDB configured CLI environment](https://v2.docs.influxdata.com/v2.0/reference/cli/influx/config/):
+_*This URL will not work for this yet.*_  Verify after pull request is complete.
```
Comment (russorat) [inline endpoint-security-state/README.md:30]:
```suggestion
```

File: endpoint-security-state/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+```
+influx apply --template-url https://raw.githubusercontent.com/influxdata/community-templates/master/endpoint-security-state/endpoint-security-state.yml
+```
+Pull directly from my [template github repo](https://github.com/darinfisher/community-templates/tree/master/endpoint-security-state).
```
Comment (russorat) [inline endpoint-security-state/README.md:34]:
```suggestion

```

File: endpoint-security-state/endpoint-security-state.yml [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            visible: true
+          - displayName: Available
+            fieldName: Available
+            visible: true
+          - displayName: Certificate
+            fieldName: Certificate
+            visible: true
+        height: 4
+        kind: Table
+        name: Name this Cell
```
Comment (russorat) [inline endpoint-security-state/endpoint-security-state.yml:70]:
```suggestion
        name: Endpoints State
```

File: endpoint-security-state/endpoint-security-state.yml [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+          [[processors.regex.tags]]
+            key = "source"
+            pattern = "^(.*)$"
+            replacement = "${1}"
+            result_key = "Endpoint"
+          [[processors.regex.tags]]
+            key = "server"
+            pattern = "^(.*)$"
+            replacement = "${1}"
+            result_key = "Endpoint"
```
Comment (russorat) [inline endpoint-security-state/endpoint-security-state.yml:240]: I think you're missing the bucket variable.

Comment (darinfisher) [reply endpoint-security-state/endpoint-security-state.yml:269]: Good catch, thank you!
It has now been added.

**Comment by darinfisher** [comment]
> Awesome @darinfisher ! Can the the source endpoints (currently example URLs) be parameterized with environment variables?

Not sure how I would do this?  Can you point me to an example?

**Pull request merged**